### PR TITLE
 XMLPoke tweaks

### PR DESCRIPTION
**Pull request by SharpeRAD**
* Extra Aliases without setting parameter
* Fix locking bug writing to the stream being read

### CODE REVIEW

**Comment by SharpeRAD** [comment]
Could do with some rebase help as I don't know how I've included your commit history?

**Comment by patriksvensson** [comment]
@SharpeRAD I'm far from a Git expert but I to be honest I think it would be easier if you [resynced your fork](https://help.github.com/articles/syncing-a-fork/) against `cake-build/cake` and created a new branch from that where you add your changes and submit a new PR.

Anyone else have an idea?

**Comment by gep13** [comment]
@patriksvensson the process described here: https://github.com/chocolatey/choco/blob/master/CONTRIBUTING.md#respond-to-feedback-on-pull-request should be all that is required.

**Comment by RichiCoder1** [comment]
That'd what I recommend too. There's also some way to rebase your branch on top of cake-build/cake/develop.

@SharpeRAD I'd leave out the `file.Open` changes, as I resolve them already in a separate pr. Otherwise looks good.

**Comment by SharpeRAD** [comment]
OK all sorted :smile:

**Comment by patriksvensson** [comment]
@RichiCoder1 You probably have the most experience with XmlPoke; would you mind doing a code review of this? :smile:

File: src/Cake.Common/Xml/XmlPokeAliases.cs [hunk 1]
```diff
@@ -122,25 +223,14 @@ public static void XmlPoke(this ICakeContext context, FilePath filePath, string
...
             var file = context.FileSystem.GetFile(filePath);
             if (!file.Exists)
             {
                 throw new FileNotFoundException("Source File not found.", file.Path.FullPath);
             }
 
-            using (var fileReader = file.Open(FileMode.Open, FileAccess.ReadWrite, FileShare.None))
-            using (var xmlReader = XmlReader.Create(fileReader))
-            using (var xmlWriter = XmlWriter.Create(fileReader))
+            using (var xmlReader = XmlReader.Create(file.Open(FileMode.Open, FileAccess.Read, FileShare.ReadWrite)))
```
Comment (RichiCoder1) [inline src/Cake.Common/Xml/XmlPokeAliases.cs]: These changes will clash with https://github.com/RichiCoder1/cake/blob/feature/XmlPokeTests/src/Cake.Common/Xml/XmlPokeAliases.cs#L141

File: src/Cake.Common/Xml/XmlPokeAliases.cs [hunk 2]
```diff
@@ -308,6 +486,7 @@ private static void XmlPoke(XmlReader source, XmlWriter destination, string xpat
             var document = new XmlDocument();
             document.PreserveWhitespace = settings.PreserveWhitespace;
             document.Load(source);
+            source.Close();
```
Comment (RichiCoder1) [inline src/Cake.Common/Xml/XmlPokeAliases.cs]: Ditto as above. You may have the correct solution over mine though.

**Comment by SharpeRAD** [comment]
Both are correct but I prefer the order of yours :smile: as you only have one stream open at a time.

**Comment by RichiCoder1** [comment]
LGTM :+1:

**Comment by patriksvensson** [comment]
OK, finally have some time to work on Cake so I'll merge this then :smile:

**Comment by patriksvensson** [comment]
Merged! :+1:

**Comment by SharpeRAD** [comment]
YAY :cake:

**Pull request closed without merging**